src/Workspaces/Remote/ServiceHub/Services/ProcessTelemetry/RemoteProcessTelemetryService.PerformanceReporter.cs

```diff
@@ -2,107 +2,76 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.Notification;
 using Microsoft.CodeAnalysis.Remote.Diagnostics;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
-using Microsoft.CodeAnalysis.SolutionCrawler;
 using Microsoft.VisualStudio.Telemetry;
+using Roslyn.Utilities;
 using RoslynLogger = Microsoft.CodeAnalysis.Internal.Log.Logger;
 
-namespace Microsoft.CodeAnalysis.Remote
+namespace Microsoft.CodeAnalysis.Remote;
+
+internal partial class RemoteProcessTelemetryService
 {
-    internal partial class RemoteProcessTelemetryService
+    /// <summary>
+    /// Track when last time report has sent and send new report if there is update after given internal
+    /// </summary>
+    private sealed class PerformanceReporter
     {
-        /// <summary>
-        /// Track when last time report has sent and send new report if there is update after given internal
-        /// </summary>
-        private class PerformanceReporter : GlobalOperationAwareIdleProcessor
-        {
-            private readonly SemaphoreSlim _event;
+        private readonly IPerformanceTrackerService _diagnosticAnalyzerPerformanceTracker;
+        private readonly TelemetrySession _telemetrySession;
+        private readonly AsyncBatchingWorkQueue _workQueue;
 
-            private readonly IPerformanceTrackerService _diagnosticAnalyzerPerformanceTracker;
-            private readonly TelemetrySession _telemetrySession;
+        public PerformanceReporter(
+            TelemetrySession telemetrySession,
+            IPerformanceTrackerService diagnosticAnalyzerPerformanceTracker,
+            CancellationToken shutdownToken)
+        {
+            _telemetrySession = telemetrySession;
+            _diagnosticAnalyzerPerformanceTracker = diagnosticAnalyzerPerformanceTracker;
 
-            public PerformanceReporter(
-                TelemetrySession telemetrySession,
-                IPerformanceTrackerService diagnosticAnalyzerPerformanceTracker,
-                IGlobalOperationNotificationService globalOperationNotificationService,
-                CancellationToken shutdownToken)
-                : base(
-                    AsynchronousOperationListenerProvider.NullListener,
-                    globalOperationNotificationService,
-                    backOffTimeSpan: TimeSpan.FromMinutes(2),
-                    shutdownToken)
-            {
-                _event = new SemaphoreSlim(initialCount: 0);
+            _workQueue = new AsyncBatchingWorkQueue(
```

Comment:
instead of all this complex logic to say "wait a couple of minutes to report remote telemetry", now we just report it every 2 minutes like before.